src/praisonai-agents/praisonaiagents/managed/_sandbox_adapter.py

```
@@ -0,0 +1,204 @@
+
+    async def execute(self, instance_id: str, command: str,
+                      timeout: Optional[int] = 300) -> Dict[str, Any]:
+        if self._sandbox is None:
+            raise RuntimeError(
+                f"{self._type} sandbox is not running; provision() was not called"
+            )
+        # The bridged execute_command tool hands us a shell command string, and
+        # models write pipes, redirects and && constantly. Without shell=True
+        # `echo x > f` is argv-split and "> f" arrives as a literal argument.
+        try:
+            result = await self._sandbox.run_command(command, shell=True)
```

Ruff (0.16.1)

[error] Function call with `shell=True` parameter identified, security issue

(S604)